lib/Site.js

```diff
@@ -25,26 +25,23 @@ const SITE_ASSET_FOLDER_NAME = 'asset';
 const TEMPLATE_ROOT_FOLDER_NAME = 'template';
 const TEMPLATE_SITE_ASSET_FOLDER_NAME = 'markbind';
 const USER_VARIABLES_PATH = '_markbind/variables.md';
+const TITLE_PREFIX_SEPARATOR = ' - ';
 
 const SITE_CONFIG_DEFAULT = {
   baseUrl: '',
-  pages: [
-    {
-      src: 'index.md',
-      title: 'Hello World',
-    },
-  ],
+  addressable: ['**/index.md'],
```

Comment:
1. Yes. Instead, the author can specify pages that are meant to be addressable, without having to list the full path of every page that's meant to be addressable.

2. Yes it does. Why is that bad?

Reply:
1. You didn't clearly state this breaking change. 😕 (cc: @Gisonrg)

2. Not sure what complications there might be.

Reply:
So this will break #178, #188 and #190...

Why wasn't this designed to be non-breaking? 😢

Reply:
#188 and #190 can be migrated to the new format using https://github.com/danielbrzn/markbind-migrate.

As for #178, I have taken a look at it and found out the areas that have to be changed should this PR be merged.

Reply:
That's irreversible, right?

I'm not convinced if that's an improvement. It's unnecessarily restrictive, especially since we're reverse-generating **siteData.json** in the existing `pages` format.

I'm inclined to this:

> 3. I think this is a better design: https://github.com/angular/angular-cli/wiki/stories-asset-configuration
We can change this later. To me, `pages` is still more intuitive and consistent with MarkBind's design.

```js
pages: [
  {
    src: 'index.md',
    title: 'Hello World'
  },
  {
    glob: '**/index.md'
  }
]
```

Reply:
> The main idea of having front matter was to ease author burden because it's a pain to have to specify every addressable page, together with its title inside `site.json`.

I believe that's solved by glob rather than Front Matter, right? Front Matter simply lets you put the `title` in the file itself, instead of in one place. That's not necessarily better, definitely not worth a breaking change in usability.  (*PR description still doesn't mention breaking change.*)

> As to the glob suggestion, would we be using the existing site.json, and then specifying a glob to collect pages that are not specified by the author?

Yes, as proposed above and expanded on here with `titlePrefix`:

```js
pages: [
  {
    src: 'index.md',
    title: 'Hello World'
  },
  {
    glob: '**/index.md',
    titlePrefix: 'CS2103/T Website'
  },
  {
    glob: '**/projectList.md',
    titlePrefix: 'CS2103/T Website - Project'
  }
]
```

As it is, I'm opposed to merging this.

Reply:
>I believe that's solved by glob rather than Front Matter, right? Front Matter simply lets you put the title in the file itself, instead of in one place. That's not necessarily better, definitely not worth a breaking change in usability. (PR description still doesn't mention breaking change.)

Yes.

To confirm, we'll generate `siteData.json` from `site.json` which is able to specify globs and src. Are titles still to be specified within `site.json`?

Also, `titlePrefix` was meant to be a global attribute, should we be supporting it on a per-glob basis?